pkg/process/exec_conf.go

```diff
@@ -103,6 +111,9 @@ func saveConfig(flagset *pflag.FlagSet, outfile string, overrides map[string]int
 			fmt.Fprintf(w, "# %s\n", f.Usage)
 		}
 		fmt.Fprintf(w, "%s: ", k)
+		if readBoolAnnotation(f, "mandatory") && (value == "") {
```

Comment:
Are you sure this works as intended on types where the default values is not empty string (EG: bool)?

Reply:
Also, does this logic work only during 'setup' (YAML creation), or always?

Reply:
It works only during setup from what I understand from the code:
		if readBoolAnnotation(f, "setup") {
			continue
		}

Reply:
And the variable "value" is always a string, it is obtained this way: 
		value := f.Value.String()

Reply:
It should technically be possible to run our programs without first running setup.  all you'd do is supply all the configuration parameter via the command line instead of via YAML.  I don't think your code overs that use case.

Reply:
It should technically be possible to run our programs without first running setup. all you'd do is supply all the configuration parameter via the command line instead of via YAML. I don't think your code overs that use case.